ecs/states-inventory-services/event-generator/event_generator.py

```diff
@@ -0,0 +1,193 @@
+    if is_linux:
+        state = random.choice(["active", "inactive", "failed"])
+        sub_state = random.choice(["running", "dead", "exited"])
+    else:
+        state = random.choice(["RUNNING", "STOPPED"])
+        sub_state = None
+
+    name = random.choice(["nginx", "sshd", "cron", "wuauserv", "winlogon"])
+    service_data = {
+        "id": name,                      # Matches ECS/osquery
+        "name": name.capitalize(),
+        "description": f"{name} service",
+        "state": state,
+        "sub_state": sub_state,
+        "address": (
+            f"/lib/{name}.so" if is_linux else f"C:\\Windows\\System32\\{name}.dll"
+        ),
+        "type": random.choice(["OWN_PROCESS", "WIN32_SHARE_PROCESS"]) if not is_linux else "system",
+        "exit_code": random.randint(0, 5) if state not in ["RUNNING", "active"] else 0,
+        "win32_exit_code": (
+            random.randint(0, 5) if not is_linux and state == "STOPPED" else 0
+        ),
+        "enabled": (
+            random.choice(["enabled", "disabled", "static"]) if is_linux else None
+        ),
+        "following": (
+            random.choice(["none", "multi-user.target"]) if is_linux else None
+        ),
+        "object_path": (
+            f"/org/freedesktop/{name}" if is_linux else None
+        ),
+        "start_type": (
+            random.choice(["AUTO_START", "DEMAND_START", "DISABLED"]) if not is_linux else None
+        ),
+        "target": {
+            "ephemeral_id": str(random.randint(1000, 9999)),
+            "type": random.choice(["start", "stop"]),
+            "address": (
+                f"/systemd/job/{name}" if is_linux else f"C:\\Jobs\\{name}"
+            )
+        }
+    }
+    return service_data
+
+def generate_user(is_linux=True):
+    return {
+        "name": random.choice(["root", "SYSTEM", "nginx", "admin", "service"])
+        if is_linux else random.choice(["SYSTEM", "Administrator", "LocalService"])
+    }
+
+def generate_wazuh():
+    return {
+        "cluster": {
+            "name": random.choice(["cluster-alpha", "cluster-beta"]),
+            "node": random.choice(["node-1", "node-2", "node-3"])
+        },
+        "schema": {
+            "version": f"{random.randint(1,3)}.{random.randint(0,9)}"
+        }
+    }
+
+def generate_random_data(number):
+    data = []
+    for _ in range(number):
+        is_linux = random.choice([True, False])
+        service_data = generate_service(is_linux=is_linux)
+        event_data = {
+            "agent": generate_agent(),
+            "file": generate_file(is_linux=is_linux),
+            "process": generate_process(is_linux=is_linux, state=service_data["state"]),
+            "service": service_data,
+            "user": generate_user(is_linux=is_linux),
+            "wazuh": generate_wazuh()
+        }
+        data.append(event_data)
+    return data
+
+def inject_events(ip, port, index, username, password, data):
+    url = f"https://{ip}:{port}/{index}/_doc"
```

Comment:
The protocol could be parametrized like it is on the [`wazuh-indexer-plugins`](https://github.com/wazuh/wazuh-indexer-plugins/tree/main/ecs)'s event generators ([example](https://github.com/wazuh/wazuh-indexer-plugins/blob/2760431a2b4f51b6e28939785511a6b1c2a7b2cb/ecs/states-inventory-groups/event-generator/event_generator.py#L91))